### ISSUE TITLE
Using state_machine breaks ActiveRecord::Base.new constructor

### ISSUE THREAD

**Issue by obrie** [pluginaweek/state_machine#63]
**Gareth Adams** opened this issue

Now, I don&rsquo;t really think this is a bug in state_machine because it seems pretty elementary, but using any kind of state_machine in my AR models seem to stop attributes being assigned when set through the class constructor.

The only thing I&rsquo;m adding is the state_machine line - and I get the same result if I add states/events too:

 @@@ ruby
    ## Before
    class ActivityBooking < ActiveRecord::Base

```
  attr_accessible :venue_id, :cyear, :cweek
  validates_presence_of :venue_id, :cyear, :cweek

  belongs_to :venue
  has_many :venue_bookings  
end

>> ActivityBooking.new(:cyear => 2010, :cweek => 27)
=> #<ActivityBooking id: nil, venue_id: nil, cyear: 2010, cweek: 27, created_at: nil, updated_at: nil, booking_confirmation_state: nil>


## After
class ActivityBooking < ActiveRecord::Base

  attr_accessible :venue_id, :cyear, :cweek
  validates_presence_of :venue_id, :cyear, :cweek

  belongs_to :venue
  has_many :venue_bookings

  state_machine :booking_confirmation_state do
  end

end


>> ActivityBooking.new(:cyear => 2010, :cweek => 27)
=> #<ActivityBooking id: nil, venue_id: nil, cyear: nil, cweek: nil, created_at: nil, updated_at: nil, booking_confirmation_state: nil>
```

 @@@

[original LH ticket](http://pluginaweek.lighthouseapp.com/projects/13288/tickets/53)

 This ticket has 0 attachment(s).

**Comment by obrie** [comment]
**Gareth Adams** commented

Sorry, to be clear:

 @@@ ruby

> > Rails.version
> > => "2.3.5"
> >  @@@

**Comment by obrie** [comment]
Hey Gareth -

You&rsquo;re right, this is pretty elementary and it&rsquo;s unclear what&rsquo;s going on.  Can you try this is a basic Rails app with no other plugins installed and tell me if you see the same problem?  Without a better understanding of what other app environment variables are in play, it&rsquo;ll be difficult to debug the issue.

For what it&rsquo;s worth, I can&rsquo;t reproduce this issue on Ruby 1.8.7 / 1.9.1, Rails 2.3.5, and state_machine 0.9.3.

**Comment by obrie** [comment]
**Gareth Adams** commented

Hey,

Sorry, I just realised this is a machine with Ruby 1.8.5 on it. I&rsquo;ve now taken the opportunity to upgrade to 1.8.7 (which should have happened a long time ago) and everything works fine.

If you&rsquo;re still interested in looking at it despite it being an ancient version of Ruby, The problem appeared with the first model I created in a brand new Rails project with no other plugins installed.
